docs/src/guides/architecture-patterns.md

```diff
@@ -0,0 +1,92 @@
+
+Additional abstractions discussed below are composed from this foundational pattern, always allowing a developer to opt into a lower-level of abstraction, enabling composition for unforseen workflows. Additionally, working with modern build tools, this level of abstraction will result in the smallest built output, as functions are the simplest structures to [tree shake](https://developers.google.com/web/fundamentals/performance/optimizing-javascript/tree-shaking/) during a build (assuming you only import the functions your code needs!).
+
+## Manager Classes
+
+To reduce cognative load, and make it easier for developers outside the ArcGIS Hub team, we have introduced a set of Manager Classes. These classes are Entity Specific (i.e. there is a `HubProjectManager` class), and they encapsulate the standard operations for an Entity - `create`, `update`, `destroy`, `fetch` etc, as well as entity specific operations.
+
+These classes are instantiated as needed, using either an `ArcGISContext` or an `ArcGISContextManager` to provide platform and identity information. The [`ArcGISContextManager` documentation](./context.html) has additional background information.
+
+Once instantiated, you can call the exposed functions without passing in platform or identity information.
+
+```ts
+import { HubProjectManager, ArcGISContextManager } from "@esri/hub-common";
+
+// Typically your application will manage a context manager instance
+const ctxMgr = await ArcGISContextManager.create({
+  username: "casey",
+  password: "abc123",
+});
+// create a project manager
+const projectMr = HubProjectManager.init(ctxMgr);
+// get a project via it's slug
+const smithStProject = await HubProjectManager.fetch("smith-st-project");
+// update some properties
+smithStProject.state = "active";
+// use the .update method on the manager, and get a new object back
+const updated = await HubProjectManager.update(smithStProject);
+// Note: Manager's return new objects from the .update function. In order
+// for modern web frameworks to track changes in data, you will want to
+// return the result of the update function vs just the object you passed in.
+```
+
+### Working with Modern Web Frameworks
+
+All modern web frameworks (React, Angular, Vue, Ember etc) leverage some form of change tracking to determine when to re-render the UI. Typically this is done via object equality (aka `===`) or ES6 proxy objects, both of which are very efficient, but add complexity when working with Class instances. To keep things simple for all frameworks, Hub.js has enforced separation between the Manager classes and the Data. All Manager functions which make changes to the data, will return a new, updated instance of the data.
+
+### Composition via Interfaces and Functions
+
```

Comment:
This section is dense and would benefit from examples to illustrate the concepts